### PR TITLE
Add info about the log rotation on Magento Cloud environment

### PR DESCRIPTION
**Pull request by dmytro-ch**
## Purpose of this pull request

This pull request introduces the documentation for log rotation flow on the Cloud environment, including the information about achieved logs files location and other minor improvements.

## Affected DevDocs pages

- https://devdocs.magento.com/cloud/project/log-locations.html

## Links to Magento source code

N/A

### CODE REVIEW

File: src/cloud/project/log-locations.md [hunk 1]
```diff
@@ -18,6 +18,9 @@ You can view the logs from the file system, the project web UI, and the `magento
 
 -  **Magento Cloud CLI**—You can view logs using the `magento-cloud log` command.
 
+{:.bs-callout-tip}
+Automatic logs rotation, compression and removal are enabled in the cloud environment. Every type of log file has its own rotating pattern and lifetime.
```
Comment (BarnyShergold) [inline src/cloud/project/log-locations.md:22]:
```suggestion
Automatic log rotation, compression and removal are enabled in the Cloud environment. Every type of log file has its own rotating pattern and lifetime.
```

File: src/cloud/project/log-locations.md [hunk 2]
```diff
@@ -143,17 +146,32 @@ Log file            | Starter and Pro Integration | Pro Staging
 **PHP access log**  | `/var/log/php.access.log`   | `/var/log/platform/<project_id>_stg/php.access.log` | `/var/log/platform/<project_id>/php.access.log`
 **PHP FPM log**     | `/var/log/app.log`          | `/var/log/platform/<project_id>_stg/php5-fpm.log` | `/var/log/platform/<project_id>/php5-fpm.log`
 
+The application logs are being archived once per day and kept for one year. In order to access the archived logs for the particular day, it is possible to find this file by a unique ID that equals `Number of Days Ago + 1`.
```
Comment (BarnyShergold) [inline src/cloud/project/log-locations.md:149]:
```suggestion
The application logs are archived once per day and kept for one year. In order to access the archived logs for a particular day, it is possible to find the file by a unique ID that equals `Number of Days Ago + 1`.
```

File: src/cloud/project/log-locations.md [hunk 3]
```diff
@@ -143,17 +146,32 @@ Log file            | Starter and Pro Integration | Pro Staging
...
+For example, the PHP access log for 21 days back on the Pro production environment can found by the following path:
```
Comment (BarnyShergold) [inline src/cloud/project/log-locations.md:150]:
```suggestion
For example, the PHP access log for 21 days in the past, on the Pro production environment can found by the following path:
```

File: src/cloud/project/log-locations.md [hunk 4]
```diff
@@ -143,17 +146,32 @@ Log file            | Starter and Pro Integration | Pro Staging
...
+
+```terminal
+/var/log/platform/<project_id>/php.access.log.22.gz
+```
+
+The archived log files are always stored in the same directory where the original was file before compression.
```
Comment (BarnyShergold) [inline src/cloud/project/log-locations.md:156]:
```suggestion
The archived log files are always stored in the same directory where the original file was located, before compression.
```

File: src/cloud/project/log-locations.md [hunk 5]
```diff
@@ -143,17 +146,32 @@ Log file            | Starter and Pro Integration | Pro Staging
...
+
+{:.bs-callout-info}
+**Deploy** and **Post-deploy** log files are not being rotated and archived. The entire deployment history can be found in the same file.
```
Comment (BarnyShergold) [inline src/cloud/project/log-locations.md:159]:
```suggestion
**Deploy** and **Post-deploy** log files are not rotated and archived. The entire deployment history can be found in the same file.
```

File: src/cloud/project/log-locations.md [hunk 6]
```diff
@@ -143,17 +146,32 @@ Log file            | Starter and Pro Integration | Pro Staging
...
 
 -  **Redis log**: `/var/log/platform/<project_id>_stg/redis-server-<project_id>_stg.log`
 -  **Elasticsearch log**: `/var/log/elasticsearch/elasticsearch.log`
+-  **Java garbage collection log**: `/var/log/elasticsearch/gc.log`
 -  **Mail log**: `/var/log/mail.log`
 -  **MySQL error log**: `/var/log/mysql/mysql-error.log`
 -  **MySQL slow log**: `/var/log/mysql/mysql-slow.log`
 -  **RabbitMQ log**: `/var/log/rabbitmq/rabbit@host1.log`
 
+Service logs are being archived and saved for different periods of time depending on the log type. The MySQL logs, having the shortest lifetime period, are removed after 7 days.
```
Comment (BarnyShergold) [inline src/cloud/project/log-locations.md:173]:
```suggestion
Service logs are archived and saved for different periods of time depending on the log type. The MySQL logs, having the shortest lifetime period, are removed after 7 days.
```

**Review by BarnyShergold — CHANGES_REQUESTED** [review]
Very nice addition!
Just a few grammar corrections

**Comment by dmytro-ch** [comment]
Hello @BarnyShergold,
Thank you for the review!
I've applied the requested changes.

File: src/cloud/project/log-locations.md [hunk 7]
```diff
@@ -18,6 +18,9 @@ You can view the logs from the file system, the project web UI, and the `magento
 
 -  **Magento Cloud CLI**—You can view logs using the `magento-cloud log` command.
 
+{:.bs-callout-tip}
+Automatic log rotation, compression and removal are enabled in the Cloud environment. Every type of log file has its own rotating pattern and lifetime.
```
Comment (bdenham) [inline src/cloud/project/log-locations.md:22]:
```suggestion
Automatic log rotation, compression, and removal are enabled in the Cloud environment. Every type of log file has its own rotating pattern and lifetime.
```

File: src/cloud/project/log-locations.md [hunk 8]
```diff
@@ -143,17 +146,32 @@ Log file            | Starter and Pro Integration | Pro Staging
 **PHP access log**  | `/var/log/php.access.log`   | `/var/log/platform/<project_id>_stg/php.access.log` | `/var/log/platform/<project_id>/php.access.log`
 **PHP FPM log**     | `/var/log/app.log`          | `/var/log/platform/<project_id>_stg/php5-fpm.log` | `/var/log/platform/<project_id>/php5-fpm.log`
 
+The application logs are archived once per day and kept for one year. In order to access the archived logs for a particular day, it is possible to find the file by a unique ID that equals `Number of Days Ago + 1`.
+For example, the PHP access log for 21 days in the past, on the Pro production environment can found by the following path:
```
Comment (bdenham) [inline src/cloud/project/log-locations.md:150]:
```suggestion
For example, the PHP access log for 21 days in the past, on the Pro production environments, is stored and named as follows:
```

File: src/cloud/project/log-locations.md [hunk 9]
```diff
@@ -143,17 +146,32 @@ Log file            | Starter and Pro Integration | Pro Staging
...
+
+```terminal
+/var/log/platform/<project_id>/php.access.log.22.gz
+```
+
+The archived log files are always stored in the same directory where the original file was located, before compression.
```
Comment (bdenham) [inline src/cloud/project/log-locations.md:156]:
```suggestion
The archived log files are always stored in the directory where the original file was located before compression.
```

File: src/cloud/project/log-locations.md [hunk 10]
```diff
@@ -143,17 +146,32 @@ Log file            | Starter and Pro Integration | Pro Staging
 **PHP access log**  | `/var/log/php.access.log`   | `/var/log/platform/<project_id>_stg/php.access.log` | `/var/log/platform/<project_id>/php.access.log`
 **PHP FPM log**     | `/var/log/app.log`          | `/var/log/platform/<project_id>_stg/php5-fpm.log` | `/var/log/platform/<project_id>/php5-fpm.log`
 
+The application logs are archived once per day and kept for one year. In order to access the archived logs for a particular day, it is possible to find the file by a unique ID that equals `Number of Days Ago + 1`.
```
Comment (bdenham) [inline src/cloud/project/log-locations.md:149]:
```suggestion
The application logs are compressed and archived once per day and kept for one year. The compressed logs are named using a unique ID that corresponds to the `Number of Days Ago + 1`.
```

File: src/cloud/project/log-locations.md [hunk 11]
```diff
@@ -143,17 +146,32 @@ Log file            | Starter and Pro Integration | Pro Staging
...
+For example, the PHP access log for 21 days in the past, on the Pro production environment can found by the following path:
+
+```terminal
+/var/log/platform/<project_id>/php.access.log.22.gz
+```
+
+The archived log files are always stored in the same directory where the original file was located, before compression.
+
+{:.bs-callout-info}
+**Deploy** and **Post-deploy** log files are not rotated and archived. The entire deployment history can be found in the same file.
```
Comment (bdenham) [inline src/cloud/project/log-locations.md:159]:
```suggestion
**Deploy** and **Post-deploy** log files are not rotated and archived. The entire deployment history is written within those log files.
```

File: src/cloud/project/log-locations.md [hunk 12]
```diff
@@ -143,17 +146,32 @@ Log file            | Starter and Pro Integration | Pro Staging
...
 
 -  **Redis log**: `/var/log/platform/<project_id>_stg/redis-server-<project_id>_stg.log`
 -  **Elasticsearch log**: `/var/log/elasticsearch/elasticsearch.log`
+-  **Java garbage collection log**: `/var/log/elasticsearch/gc.log`
 -  **Mail log**: `/var/log/mail.log`
 -  **MySQL error log**: `/var/log/mysql/mysql-error.log`
 -  **MySQL slow log**: `/var/log/mysql/mysql-slow.log`
 -  **RabbitMQ log**: `/var/log/rabbitmq/rabbit@host1.log`
 
+Service logs are archived and saved for different periods of time depending on the log type. The MySQL logs, having the shortest lifetime period, are removed after 7 days.
```
Comment (bdenham) [inline src/cloud/project/log-locations.md:173]:
```suggestion
Service logs are archived and saved for different periods of time, depending on the log type. For example, MySQL logs have the shortest lifetime—removed after 7 days.
```

**Review by bdenham — CHANGES_REQUESTED** [review]
Just need a few wording changes. Great additions!

**Comment by dmytro-ch** [comment]
Hello @bdenham,
Thank you for your review!
I've applied the requested changes.

**Review by bdenham — APPROVED** [review]
Looks good! Thanks for the update!

**Comment by bdenham** [comment]
@magento import code to magento-devdocs/devdocs

**Comment by magento-engcom-team** [comment]
@bdenham the branch with code successfully imported into`magento-devdocs/devdocs` repository. Branch name: `imported-magento-devdocs-8970`.

**Pull request merged**